clients/lsp-rust.el

```diff
@@ -910,20 +910,20 @@ meaning."
   (if (and (lsp-rust-analyzer-initialized?)
            (eq buffer (current-buffer)))
       (lsp-request-async
-       "rust-analyzer/inlayHints"
+       "experimental/inlayHints"
        (lsp-make-rust-analyzer-inlay-hints-params
         :text-document (lsp--text-document-identifier))
        (lambda (res)
          (remove-overlays (point-min) (point-max) 'lsp-rust-analyzer-inlay-hint t)
          (dolist (hint res)
-           (-let* (((&rust-analyzer:InlayHint :range :label :kind) hint)
-                   ((&RangeToPoint :start :end) range)
-                   (overlay (make-overlay start end nil 'front-advance 'end-advance)))
+           (-let* (((&rust-analyzer:InlayHint :position :label :kind) hint)
+                   (pos (lsp--position-to-point position))
+                   (overlay (make-overlay pos (+ pos 1) nil 'front-advance 'end-advance)))
```

Comment:
I'm not actually sure what that second parameter should be; `pos` doesn't work, but the position we get doesn't really have an _end_ per se.

Reply:
My guess would be that `pos` doesn't work because we set `evaporate`, which makes Emacs remove the hints when they have length 0. I don't know if that's necessary, since we also remove the overlays manually when we update them. Maybe @yyoncho knows more here